core/src/main/java/jenkins/tasks/SimpleBuildWrapper.java

```diff
@@ -117,16 +182,70 @@ public void setDisposer(@NonNull Disposer disposer) {
      * Must be safely serializable, so it receives runtime context comparable to that of the original setup.
      */
     public static abstract class Disposer implements Serializable {
+
+        private Boolean wrapperRequiresWorkspace;
+
+        /**
+         * Determines whether or not this end-of-wrapped-block callback requires a workspace context (working
+         * directory and launcher).
+         * <p>
```

Comment:
Similarly here.

Reply:
Not sure why this would be restricted - the caller needs this info to call the right `tearDown` overload.

Reply:
Again, the only external caller should be in `workflow-basic-steps`, so this is a reminder that “regular” plugins should not touch it.

Reply:
Oh, so `workflow-basic-steps` is already using a `useBeta` flag? Fair enough then.